backend/de.metas.business/src/main/java/org/eevolution/api/impl/ProductBOMVersionsDAO.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package org.eevolution.api.impl;
+
+import de.metas.organization.OrgId;
+import de.metas.product.ProductId;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.eevolution.api.BOMVersionsCreateRequest;
+import org.eevolution.api.ProductBOMVersionsId;
+import org.eevolution.model.I_PP_Product_BOMVersions;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+
+@Repository
+public class ProductBOMVersionsDAO
+{
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+
+	@NonNull
+	public ProductBOMVersionsId createBOMVersions(@NonNull final BOMVersionsCreateRequest request)
+	{
+		final OrgId orgId = request.getOrgId();
+
+		final I_PP_Product_BOMVersions bomVersionsRecord = newInstance(I_PP_Product_BOMVersions.class);
+		bomVersionsRecord.setAD_Org_ID(orgId.getRepoId());
+		bomVersionsRecord.setM_Product_ID(request.getProductId().getRepoId());
+		bomVersionsRecord.setName(request.getName());
+		bomVersionsRecord.setDescription(request.getDescription());
+
+		saveRecord(bomVersionsRecord);
+
+		return ProductBOMVersionsId.ofRepoId(bomVersionsRecord.getPP_Product_BOMVersions_ID());
+	}
+
+	@NonNull
+	public Optional<ProductBOMVersionsId> retrieveBOMVersionsId(final ProductId productId)
```

Comment:
both products and boms are per Client & Org, meaning you can e.g. have a product in Org=* and a bom(-version) with that product in org=A.
And then org=B might want to create their own BOM(-version). 
So i think it's important to include the OrgID when searching for a bomversion.
The lookup should be with `AD_Org_ID in (the-org, OrgId.any)`, ofc prefering "the-org".

Also, AFAIS the BOM should alsways have the same AD_Org_ID as its BOM-Version

Reply:
As talked, let's have this as a follow-up